### ISSUE TITLE
model_read_lib visitAggregation is incorrect.

### ISSUE THREAD

**Issue by matth79** [calacademy-research/COMB#78]
Setting aggregation to day does not do the right thing (per report in 6/29 meeting).

This issue tracks manual verification of each allowed setting and fixing as necessary.

**Comment by mkclapp** [comment]
Setting visitAggregation = "day" now suddenly seems to work in the script combined.R without me changing anything else. The values for the # of ML 'hits' per morning appears reasonable. If someone else could confirm, I think we can close this issue? @matth79 

The bad news is that simply changing the aggregation didn't seem to make model fit any better (MCMC chain convergence fails the Rhat < 1.1 rule-of-thumb) so we still have our work cut out for us in terms of improving model fit. Things to check (jotting down here before starting a new issue):

- prior specification in JAGS model. Right now, they're very uninformative
- reasonable values/ranges for input data. For example, making sure visit index is an integer between 1 and nvisit, and not the Julian day of the visit - it looks like this is the case for data$indices$visit.aru but not data$indices$visit.pc. (We should also make sure the date corresponding to the visit is stored somewhere, for if/when we use Julian date in the model for detection.)
- nsite for aru = 84; nsite for pc = 81. This may be causing problems in estimating anything at the site level (eg. z, p). Need to investigate whether JAGS code is "smart" enough to read the site identifiers rather than just their place in the index of sites (or if we need to be smart enough to explicitly code that)
- ...probably other things

**Comment by ddkapan** [comment]
@mschulist Can you confirm this works as @mkclapp says, if so this issue and a referring issue: can be closed.

**Comment by mschulist** [comment]
I reran the script and it does work without errors, but I cannot find the object `visitAggregation` in `combined.R`. So it works, but I could be on the incorrect branch.

**Comment by mkclapp** [comment]
Hi @mschulist,
it is an argument of the readCombined function and can simply be added to the others as such (I think/hope):

data <- readCombined(
  species = c(speciesCode),
  years = c(year),
  **visitAggregation = "day",** # the other option would be "file"
  visitLimit = aruVisitLimit,
  thresholdOptions = list(
    value = threshold,
    is.quantile = F
  ),
  squeeze = T
)

Feel free to add and push if you have an open branch for this issue!

**Comment by ddkapan** [comment]
- [ ] Looking like aggregation working, correct?

- [ ] But let's discuss and address the great stuff above in Matt's comments before closing the issue
